sources/PTYSession.m

```diff
@@ -8046,6 +8046,18 @@ - (void)screenRaise:(BOOL)flag {
     }
 }
 
+// Sets current session proxy icon.
+- (void) setProxyIcon:(NSString *)value {
+    NSURL* url = nil;
+    if (value) {
+        url = [NSURL fileURLWithPath:value];
+    }
+    self.userDesiredProxyIcon = url;
+    if (url) {
+        [[_delegate parentWindow] updateProxyIcon];
```

Comment:
Don't go right to the window controller from here. For one thing, this might not be the active session. It also adds dependencies on the window controller, and I'd like to remove as many of them as possible.

Instead, add a delegate method (like `sessionProxyIconDidChange:`) whose implementation in `PTYTab.m` will be similar to `sessionCurrentDirectoryDidChange:`: It should check if the session is the current session, and if so invoke a method on *its* delegate (like `tab:proxyIconDidChange:`). Since all that `updateCurrentLocation` does is mess with the proxy icon, may as well rename it to `updateProxyIcon`.